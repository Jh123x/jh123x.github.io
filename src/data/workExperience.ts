import DSO_Logo from "../img/logos/dso_logo.jfif";
import NUS_LOGO from "../img/logos/nus_logo.jfif";
import FORUS_LABS_LOGO from "../img/logos/foruslabs_logo.jfif";
import SHOPEE_LOGO from "../img/logos/shopee_logo.jfif";
import TIKTOK_LOGO from "../img/logos/Tiktok_Logo.png";
import { WorkExperienceEntry } from "./types";


/**
 * Latest experience at the front of the array
 */
const workExperience: WorkExperienceEntry[] = [
  {
    year: "Jun 2023 - Present",
    logo: TIKTOK_LOGO,
    title: "Backend Software Engineer",
    subTitle: "PIPO (SG) Pte Ltd",
    text: "Collaborate with multiple cross-functional teams to deliver high quality work in rapid product development. Align with product and org objectives, and coordinate with cross-functional teams on delivering key results. Responsible for designing/architect the solution and providing technical support for engineers in the team",
  },
  {
    year: "Aug 2022 - May 2023",
    logo: NUS_LOGO,
    title:
      "CS2107 (Introduction to Information Security) Assignment Teaching Assistant",
    subTitle: "National University of Singapore",
    text: "In this role, my responsibilities includes creating CTF style assignments, holding CTF consultations, answering queries and marking assignments. We work closely with the professors to ensure that the assignment questions covers the scope of the lecture and bridge the theory taught with lecture with practical applications.",
  },
  {
    year: "May 2022 - Aug 2022",
    title: "Security Research Intern",
    subTitle: "DSO National Laboratories",
    logo: DSO_Logo,
    text: "During my internship, I research on tools for automated vulnerability discovery and analysis. I also developed a proof of concept for finding vulnerabilities in binary code. These tools are useful in automatic vulnerability discovery and analysis. As this is not a commercial product, I am unable to disclose the details of the project.",
  },
  {
    year: "May 2022",
    title: "CP2106 (Orbital Project) Teaching Assistant",
    logo: NUS_LOGO,
    subTitle: "National University of Singapore",
    text: "Together with my team from NUS Greyhats, we conducted a web security workshop for the students of CP2106. It covers concepts ranging from SQL injection to XSS. We also include the source code for the CTF. It aims to promote better security in the projects by the students.",
    links: {
      github: {
        link: "https://github.com/NUSGreyhats/Orbital-2022",
        tooltip: "GitHub Repository",
      },
    },
  },
  {
    year: "Aug 2021 - Nov 2021",
    title: "Part Time Fullstack Engineering Intern",
    subTitle: "Forus Labs",
    logo: FORUS_LABS_LOGO,
    text: "During my internship, I worked on CI/CD Pipelines and flutter development for the Checkist application. The application is available in the Google Play Store.",
    links: {
      google: {
        link: "https://play.google.com/store/apps/details?id=com.foruslabs.checkist&hl=en_SG&gl=US&pli=1",
        tooltip: "Google Play Store",
      }
    },
  },
  {
    year: "Aug 2021 - Nov 2021",
    title: "CS2103T (Software Engineering Project) Senior Student Mentor",
    logo: NUS_LOGO,
    subTitle: "National University of Singapore",
    text: "In this role, my responsibilities include guiding students on technical difficulties encountered within their assignments and creating solutions.",
  },
  {
    year: "May 2021 - Jul 2021",
    title: "Software Engineering Intern (Backend)",
    subTitle: "Shopee Singapore",
    logo: SHOPEE_LOGO,
    text: "During my internship, I worked on projects related to the backend of Shopee's e-commerce platform. This includes the usage of Golang, Python2, Django and MySQL.",
  },
  {
    year: "May 2021",
    title: "CP2106 (Orbital Project) Teaching Assistant",
    logo: NUS_LOGO,
    subTitle: "National University of Singapore",
    text: "Together with my team from NUS Greyhats, we conducted a web security workshop for the students of CP2106.",
    links: {
      youtube: {
        link: "https://www.youtube.com/watch?v=yUs9zKqGJDU",
        tooltip: "YouTube Video",
      },
      github: {
        link: "https://github.com/NUSGreyhats/Orbital-2021",
        tooltip: "GitHub Repository",
      },
    },
  },
  {
    year: "Jan 2021 - May 2021",
    logo: NUS_LOGO,
    title:
      "CS2107 (Introduction to Information Security) Assignment Teaching Assistant",
    subTitle: "National University of Singapore",
    text: "In this role, my responsibilities includes creating CTF style assignments, hold consultations and marking assignments.",
  },
  {
    year: "Jan 2021 - Apr 2021",
    title: "CS2103T (Software Engineering Project) Student Mentor",
    logo: NUS_LOGO,
    subTitle: "National University of Singapore",
    text: "In this role, my responsibilities include guiding students on technical difficulties encountered within their assignments and creating solutions.",
  },
];

export default workExperience;
