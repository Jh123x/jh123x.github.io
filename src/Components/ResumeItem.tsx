import * as React from "react";
import styled from "styled-components";
import itemMap from "../data/iconMap";
import { WorkExperienceEntry } from "../data/types";

const ResumeItem = ({
  year,
  title,
  subTitle,
  text,
  logo,
  links,
}: WorkExperienceEntry) => (
  <ResumeItemStyled>
    <div className="left-content">
      <p>{year}</p>
    </div>
    <div className="right-content">
      {logo !== undefined ? (
        <img src={logo} width={100} height={100} alt={subTitle} />
      ) : (
        <></>
      )}
      <h5>{title}</h5>
      <h6>{subTitle}</h6>
      <p>{text}</p>
      {links ? itemMap(links) : <></>}
    </div>
  </ResumeItemStyled>
);

const ResumeItemStyled = styled.div`
  display: flex;
  flex-direction: row;
  @media screen and (max-width: 421px) {
    p,
    h5,
    h6 {
      font-size: 80%;
    }
  }
  &:not(:last-child) {
    padding-bottom: 3rem;
  }
  .left-content {
    width: 50%;
    padding-left: 20px;
    position: relative;
    &::before {
      content: "";
      position: absolute;
      left: -10px;
      top: 5px;
      height: 15px;
      width: 15px;
      border-radius: 50%;
      border: 2px solid var(--border-color);
      background-color: var(--background-dark-color);
    }
    p {
      display: inline-block;
    }
  }
  .right-content {
    padding-right: 5rem;
    position: relative;
    max-width: 60%;

    &::before {
      content: "";
      top: 15px;
      height: 2px;
      width: 3rem;
      background-color: var(--border-color);
    }
    h5 {
      color: var(--primary-color);
      font-size: 2rem;
      padding-bottom: 0.4rem;
    }
    h6 {
      padding-bottom: 0.6rem;
      font-size: 1.5rem;
    }
  }
`;
export default ResumeItem;
