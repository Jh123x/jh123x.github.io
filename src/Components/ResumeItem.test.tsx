import * as React from "react";
import { describe, expect, test } from "@jest/globals";
import ResumeItem from "./ResumeItem";
import workExperience from "../data/workExperience";

describe("Resume Item", () => {
  for (const experience of workExperience) {
    test(`should render a ${experience.title} item`, () => {
      const wrapper = <div>{<ResumeItem {...experience} />}</div>;
      expect(wrapper).toMatchSnapshot();
    });
  }
});
