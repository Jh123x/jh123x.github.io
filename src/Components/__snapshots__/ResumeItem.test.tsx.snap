// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume Item should render a Backend Software Engineer (Global Payments) item 1`] = `
<div>
  <ResumeItem
    logo={
      {
        "blurDataURL": "data:image/png;base64,imagedata",
        "height": 40,
        "src": "/img.jpg",
        "width": 40,
      }
    }
    subTitle="Bytedance Pte Ltd"
    text="Creating and managing user payment information for Bytedance products, maintaining APIs for payment processing, and ensuring the security of payment information."
    title="Backend Software Engineer (Global Payments)"
    year="Jun 2023 - Present"
  />
</div>
`;

exports[`Resume Item should render a CP2106 (Orbital Project) Teaching Assistant item 1`] = `
<div>
  <ResumeItem
    link={
      {
        "github": {
          "link": "https://github.com/NUSGreyhats/Orbital-2022",
          "tooltip": "GitHub Repository",
        },
      }
    }
    logo={
      {
        "blurDataURL": "data:image/png;base64,imagedata",
        "height": 40,
        "src": "/img.jpg",
        "width": 40,
      }
    }
    subTitle="National University of Singapore"
    text="Together with my team from NUS Greyhats, we conducted a web security workshop for the students of CP2106. It covers concepts ranging from SQL injection to XSS. We also include the source code for the CTF. It aims to promote better security in the projects by the students."
    title="CP2106 (Orbital Project) Teaching Assistant"
    year="May 2022"
  />
</div>
`;

exports[`Resume Item should render a CP2106 (Orbital Project) Teaching Assistant item 2`] = `
<div>
  <ResumeItem
    link={
      {
        "github": {
          "link": "https://github.com/NUSGreyhats/Orbital-2021",
          "tooltip": "GitHub Repository",
        },
        "youtube": {
          "link": "https://www.youtube.com/watch?v=yUs9zKqGJDU",
          "tooltip": "YouTube Video",
        },
      }
    }
    logo={
      {
        "blurDataURL": "data:image/png;base64,imagedata",
        "height": 40,
        "src": "/img.jpg",
        "width": 40,
      }
    }
    subTitle="National University of Singapore"
    text="Together with my team from NUS Greyhats, we conducted a web security workshop for the students of CP2106."
    title="CP2106 (Orbital Project) Teaching Assistant"
    year="May 2021"
  />
</div>
`;

exports[`Resume Item should render a CS2103T (Software Engineering Project) Senior Student Mentor item 1`] = `
<div>
  <ResumeItem
    logo={
      {
        "blurDataURL": "data:image/png;base64,imagedata",
        "height": 40,
        "src": "/img.jpg",
        "width": 40,
      }
    }
    subTitle="National University of Singapore"
    text="In this role, my responsibilities include guiding students on technical difficulties encountered within their assignments and creating solutions."
    title="CS2103T (Software Engineering Project) Senior Student Mentor"
    year="Aug 2021 - Nov 2021"
  />
</div>
`;

exports[`Resume Item should render a CS2103T (Software Engineering Project) Student Mentor item 1`] = `
<div>
  <ResumeItem
    logo={
      {
        "blurDataURL": "data:image/png;base64,imagedata",
        "height": 40,
        "src": "/img.jpg",
        "width": 40,
      }
    }
    subTitle="National University of Singapore"
    text="In this role, my responsibilities include guiding students on technical difficulties encountered within their assignments and creating solutions."
    title="CS2103T (Software Engineering Project) Student Mentor"
    year="Jan 2021 - Apr 2021"
  />
</div>
`;

exports[`Resume Item should render a CS2107 (Introduction to Information Security) Assignment Teaching Assistant item 1`] = `
<div>
  <ResumeItem
    logo={
      {
        "blurDataURL": "data:image/png;base64,imagedata",
        "height": 40,
        "src": "/img.jpg",
        "width": 40,
      }
    }
    subTitle="National University of Singapore"
    text="In this role, my responsibilities includes creating CTF style assignments, holding CTF consultations, answering queries and marking assignments. We work closely with the professors to ensure that the assignment questions covers the scope of the lecture and bridge the theory taught with lecture with practical applications."
    title="CS2107 (Introduction to Information Security) Assignment Teaching Assistant"
    year="Aug 2022 - May 2023"
  />
</div>
`;

exports[`Resume Item should render a CS2107 (Introduction to Information Security) Assignment Teaching Assistant item 2`] = `
<div>
  <ResumeItem
    logo={
      {
        "blurDataURL": "data:image/png;base64,imagedata",
        "height": 40,
        "src": "/img.jpg",
        "width": 40,
      }
    }
    subTitle="National University of Singapore"
    text="In this role, my responsibilities includes creating CTF style assignments, hold consultations and marking assignments."
    title="CS2107 (Introduction to Information Security) Assignment Teaching Assistant"
    year="Jan 2021 - May 2021"
  />
</div>
`;

exports[`Resume Item should render a Part Time Fullstack Engineering Intern item 1`] = `
<div>
  <ResumeItem
    link={
      {
        "google": {
          "link": "https://play.google.com/store/apps/details?id=com.foruslabs.checkist&hl=en_SG&gl=US&pli=1",
          "tooltip": "Google Play Store",
        },
      }
    }
    logo={
      {
        "blurDataURL": "data:image/png;base64,imagedata",
        "height": 40,
        "src": "/img.jpg",
        "width": 40,
      }
    }
    subTitle="Forus Labs"
    text="During my internship, I worked on CI/CD Pipelines and flutter development for the Checkist application. The application is available in the Google Play Store."
    title="Part Time Fullstack Engineering Intern"
    year="Aug 2021 - Nov 2021"
  />
</div>
`;

exports[`Resume Item should render a Security Research Intern item 1`] = `
<div>
  <ResumeItem
    logo={
      {
        "blurDataURL": "data:image/png;base64,imagedata",
        "height": 40,
        "src": "/img.jpg",
        "width": 40,
      }
    }
    subTitle="DSO National Laboratories"
    text="During my internship, I research on tools for automated vulnerability discovery and analysis. I also developed a proof of concept for finding vulnerabilities in binary code. These tools are useful in automatic vulnerability discovery and analysis. As this is not a commercial product, I am unable to disclose the details of the project."
    title="Security Research Intern"
    year="May 2022 - Aug 2022"
  />
</div>
`;

exports[`Resume Item should render a Software Engineering Intern (Backend) item 1`] = `
<div>
  <ResumeItem
    logo={
      {
        "blurDataURL": "data:image/png;base64,imagedata",
        "height": 40,
        "src": "/img.jpg",
        "width": 40,
      }
    }
    subTitle="Shopee Singapore"
    text="During my internship, I worked on projects related to the backend of Shopee's e-commerce platform. This includes the usage of Golang, Python2, Django and MySQL."
    title="Software Engineering Intern (Backend)"
    year="May 2021 - Jul 2021"
  />
</div>
`;
